### PR TITLE
Make dynamic member lookup writable

### PR DESCRIPTION
**Pull request by stephencelis**
Realizing we should probably make DML writable.

### CODE REVIEW

**Comment by stephencelis** [comment]
On more thought...this could break invariants, _e.g._:

```swift
nonEmptyIdentifiedArray       // NonEmpty<IdentifiedArrayOf<A>>
nonEmptyIdentifiedArray.count // 1
let element = nonEmptyIdentifiedArray[0]
nonEmptyIdentifiedArray[id: element.id] = nil
nonEmptyIdentifiedArray.      // 0
```

So if we do open up external writability...we should consider exposing unsafe/unchecked interfaces and introducing preconditions.

File: Sources/NonEmpty/NonEmpty.swift [hunk 1]
```diff
@@ -10,8 +10,9 @@ public struct NonEmpty<Collection: Swift.Collection>: Swift.Collection {
     self.rawValue = rawValue
   }
 
-  public subscript<Subject>(dynamicMember keyPath: KeyPath<Collection, Subject>) -> Subject {
-    self.rawValue[keyPath: keyPath]
+  public subscript<Subject>(dynamicMember keyPath: WritableKeyPath<Collection, Subject>) -> Subject {
+    _read { yield self.rawValue[keyPath: keyPath] }
+    _modify { yield &self.rawValue[keyPath: keyPath] }
```
Comment (KyLeggiero) [inline Sources/NonEmpty/NonEmpty.swift:15]: why `_read` and `_modify` rather than `get` and `set`?

Comment (stephencelis) [reply Sources/NonEmpty/NonEmpty.swift:15]: We prefer `_read` and `_modify` here for performance reasons, as it allows direct read/write access to the underlying value.

**Comment by stephencelis** [comment]
Going to close this for now since I don't think we intend to merge it in its current state.

**Pull request closed without merging**